packages/core/src/sagas/incomingActivitySaga.js

```diff
@@ -26,13 +26,18 @@ function* observeActivity({ directLine, userID }) {
     yield put(incomingActivity(activity));
 
     // Update suggested actions
+    // TODO: [P3] We could put this logic inside reducer to minimize number of actions dispatched
     const messageActivities = yield select(activitiesOfType('message'));
     const lastMessageActivity = messageActivities[messageActivities.length - 1];
 
     if (activityFromBot(lastMessageActivity)) {
-      const { suggestedActions: { actions } = {} } = lastMessageActivity;
+      const { suggestedActions: { actions, to } = {} } = lastMessageActivity;
 
-      yield put(setSuggestedActions(actions));
+      if (!to || !to.length || to.includes(userID)) {
```

Comment:
This can optionally be simplified:
```js
const params = (to || []).includes(userId) ? actions : null;
yield put(setSuggestedActions(params));
```

Reply:
I was thinking to simplify this too (in a single line). I thought people would hate me. Good to know you don't. 😄

The correct one should be `!to || !to.length || to.includes(userID) ? actions : null`. I will add a comment above it to explain.

I think it is better to write comment and have shorter-but-cryptic code, than no comment but longer code.

Will fix it.